include/ruby/internal/core/rdata.h

```diff
@@ -174,11 +174,14 @@ rb_cData(void)
 
 #define rb_data_object_wrap_0 rb_data_object_wrap
 #define rb_data_object_wrap_1 rb_data_object_wrap_warning
+#define rb_data_object_wrap_  rb_data_object_wrap_ /* Used here vvvv */
 #define rb_data_object_wrap   RUBY_MACRO_SELECT(rb_data_object_wrap_, RUBY_UNTYPED_DATA_WARNING)
 #define rb_data_object_get_0  rb_data_object_get
 #define rb_data_object_get_1  rb_data_object_get_warning
+#define rb_data_object_get_   rb_data_object_get_ /* Used here vvvv */
 #define rb_data_object_get    RUBY_MACRO_SELECT(rb_data_object_get_, RUBY_UNTYPED_DATA_WARNING)
 #define rb_data_object_make_0 rb_data_object_make
 #define rb_data_object_make_1 rb_data_object_make_warning
+#define rb_data_object_make_  rb_data_object_make_ /* Used here vvvv */
 #define rb_data_object_make   RUBY_MACRO_SELECT(rb_data_object_make_, RUBY_UNTYPED_DATA_WARNING)
```

Comment:
It looks like this part doesn't work as-is on Windows:
https://github.com/ruby/ruby/pull/4371/checks?check_run_id=2304484466

Reply:
Seems I choked a glitchy C preprocessor corner case.

1. `Data_Wrap_Struct(...)` expands to `rb_data_object_wrap(...)`
2. which expands to `RUBY_MACRO_SELECT(rb_data_object_wrap_, RUBY_UNTYPED_DATA_WARNING)`
3. which expands to `rb_data_object_wrap_0`
4. which expands to `rb_data_object_wrap`, so far so good, but 
5. this is a recursive macro expansion (see step 1).  Everybody stops expanding...
    - in step 3 for MSVC, and
    - in step 4 for GCC etc.

I'd like to say this is a MSVC bug but that doesn't save anything.  Looking for a workaround...

Reply:
https://gcc.godbolt.org/z/hTd8cMzjz

Reply:
https://github.com/ruby/ruby/pull/4371/commits/682f94f10a0692ad6b64e94d17ba8e0b4307ee80